# The Capital Asset Pricing Model

The CAPM setting is not much different than the one considered by Markowitz. In the CAPM setting, investors are:
- Risk-averse
- Prefer higher returns
- Are unwilling to buy anything other than the optimal portfolio

Sharp introduces the concept of a market portfolio - that is a bundle of all possible investments in the world (both bonds and stocks) - and the risk return combination of this portfolio is superior to the one of any other portfolio. The expected return of the market portfolio coincides with the expected return of the market. Because this is a diversified portfolio, it makes sense it is optimal in terms of risk. It contains no idiosyncratic risk. The only risk is systematic risk. And we would expect this portfolio lies somewhere on the efficient frontier as it is an efficient portfolio, the most efficient actually. 

So far, this looks like a simple extension of the ideas introduced by Markowitz. However, the CAPM assumes the existence of a risk-free asset - it has a positive rate of return but zero risk associated with it. This automatically means there will be people so risk averse they will prefer to buy the risk-free only. If they want to be 100% certain their investment contains zero risk, they'll be willing to accept a lower expected rate of return. 

Wait, why should we assume the risk-free has a lower expected rate of return? Under the CAPM, markets are efficient. In an efficient market, investors are compensated only for additional risk they're willing to bear. Once they own an efficient portfolio, they can't arbitrage the system and earn a higher expected return for the same level of risk. In such setting, a higher rate of return means more risk. Conversely, a lower level of risk means lower return. Zero level of risk translates to the lowest level of expected return. 

So what changes after we introduce a risk free asset then? It means the market portfolio isn't the only asset rational investors would invest in. The risk free investment offers something new to investors. Zero risk. So, rational investors will form their portfolios considering both the risk free rate and the market portfolio. 

How much are they going to invest in risk free and how much are they going to invest in the market portfolio? Depends on how much they want to earn.

The line that connects the risk free rate and is the tangent to the efficient frontier is called the Capital Market Line. 
**The point where the CAPM Line intersects the efficient frontier is the market portfolio**. 

So, every investor can invest in a combo between the risk free asset and the market portfolio according to their risk apetite. If they're willing to risk more, they'll hold a greater portion of the market portfolio. In the opposite case, if they're unwilling to risk, they'll buy more of the risk free asset and less of the market portfolio.

## Understanding and calculating a security's Beta

> Beta helps us quantify the relationship between a security and the overall market portfolio.

If there is a crisis in the economy, it is reasonable to expect the prices of most assets that make up the market portfolio will decrease and the market portfolio will experience a negative rate of return (let's say -5%). Investors can't protect against this via diversification because this is a systemic risk. 

However, some securities in the market portfolio are less risky. They have a lower sd and will decrease less than the market. 

For example, stock A is less volatile and only loses 3%. On the other hand, stock B has a high level of volatility and loses 7% of its value. When the economyu recovers, the market portfolio will do well and earns 9%. Stock A, the low risk low return investment, will earn investors 5% while stock B, the riskier stock, will earn 12%. 

This shows us that stocks can have a different behavior regarding overall market performance. Some stocks are safer (lose less, earn less) while other stocks are riskier (lose more, earn more). This is where beta comes in handy. It allows us to measure the relationship between a stock and the market portfolio.

$$\beta = \frac{cov(r_x, r_m)}{\sigma ^2_m}$$

Beta can be calculated as the covariance between the stock and the market divided by the variance of the market. **It measures the market risk that cannot be avoided through diversification**. The riskier the stock, the higher its beta. 

- Beta = 0: No relationship
- Beta < 1: Defensive (if market does poorly, these will lose less)
- Beta > 1: Aggressive (does better than market if econ is good, lose more when it goes down)

E.g. Walmart beta = 0.09. Ford beta = 1.1

## Calculating the Beta of a Stock

Beta is a fraction. 

$$\beta = \frac{cov(PG, mkt)}{var(mkt)}$$

The nominator equals the covariance between the stock and the market. The denominator is the variance of the market. Beta is typically measured with data from the past 5 years. 

In [1]:
import numpy as np
import pandas as pd
from pandas_datareader import data as wb

tickers = ['PG', '^GSPC']
data = pd.DataFrame()
for t in tickers:
    data[t] = wb.DataReader(t, data_source='yahoo', start='2015-7-15')['Adj Close']

/opt/anaconda3/lib/python3.7/site-packages/pandas_datareader/compat/__init__.py:7: FutureWarning: pandas.util.testing is deprecated. Use the functions in the public API at pandas.testing instead.
  from pandas.util.testing import assert_frame_equal


In [2]:
# first, lets get log returns
sec_returns = np.log(data/data.shift(1))
sec_returns.tail()

PG     ^GSPC
Date                          
2020-07-09 -0.003342 -0.005660
2020-07-10  0.011446  0.010412
2020-07-13  0.001291 -0.009407
2020-07-14  0.008349  0.013317
2020-07-15  0.003232  0.009022

In [3]:
# next, we create a cov matrix b/t PG and the market
cov = sec_returns.cov() * 250
cov

PG     ^GSPC
PG     0.040957  0.024282
^GSPC  0.024282  0.037681

In [4]:
# then, iloc helps us obtain the cov between the stock and the market as a float
cov_with_market = cov.iloc[0, 1]
cov_with_market

0.0242815464313929

Alright, the nominator is ready. Nominator = variance of the market. This means we can use a new variable called `market_var` containing the annualized variance of the S&P500.

In [5]:
market_var = sec_returns['^GSPC'].var() * 250
market_var

0.03768116597734568

Now, we're ready to calc beta. 

### Beta:
$$\beta_{pg} = \frac{\sigma_{pg,m}}{\sigma_m}$$

In [7]:
PG_beta = cov_with_market / market_var
PG_beta

0.6443947739300644

This value is 0 < x < 1. This means that PG is a defensive stock. When the market goes up, PG's equity value won't skyrocket, while in times of crisis you'll lose less. It's good practice to verify your calculations of such indicators were correct. [barchart.com](https://www.barchart.com/stocks/quotes/PG/overview)
(barchart is saying beta is 0.42 yet we get 0.64 here, why?)